src/renderers/shared/utils/ReactErrorUtils.js

```diff
@@ -155,33 +155,40 @@ if (__DEV__) {
       e,
       f,
     ) {
-      ReactErrorUtils._hasCaughtError = false;
-      ReactErrorUtils._caughtError = null;
-
       depth++;
-      const thisDepth = depth;
+
+      let error;
```

Comment:
If the onerror event doesn't happen (e.g. because browser flakiness, or monkey patched DOM) then `error` will be `undefined` and that will be the error that we catch.

I think that's a fine value for that case but maybe we should explicitly set that here so that we know that it's intentional and that it can have that value.

E.g.

```js
let error = undefined;
```